primitives/trie/src/cache/mod.rs

```diff
@@ -86,21 +328,13 @@ impl CacheSize {
 pub struct LocalTrieCache<H: Hasher> {
 	/// The shared trie cache that created this instance.
 	shared: SharedTrieCache<H>,
+
 	/// The local cache for the trie nodes.
-	node_cache: Mutex<HashMap<H::Out, NodeOwned<H::Out>>>,
-	/// Keeps track of all the trie nodes accessed in the shared cache.
-	///
-	/// This will be used to ensure that these nodes are brought to the front of the lru when this
-	/// local instance is merged back to the shared cache.
-	shared_node_cache_access: Mutex<HashSet<H::Out>>,
```

Comment:
It's unnecessary since the node cache now works more like a traditional cache, that is: values from the higher level cache (the shared cache) are added to the local cache (the lower cache), or in other words now the local cache will contain the values which were tracked here, so it's not necessary to track them separately anymore.

Reply:
I see, could it also be done for the value cache the same way?

Reply:
> I see, could it also be done for the value cache the same way?

Technically yes, I guess.

I just didn't do it because it didn't seem to be necessary, so I just left it as-is.

And also, IIRC the local cache will still get automatically updated by the trie itself if the value in the shared cache is stale.

Reply:
tbh I would just not do the reordering/access recording for shared cache and just make it on shared cache access (we exclusive lock anyway), sounds more straightforward to me, but I did not check the actual impact and I may be missing something. Content fetch on db backend will be more prioritized in cache but I guess it would really be an issue only if cache is very small.

Reply:
Looks a bit awkward to have different mechanism in both case, but it is true that it is different footprints.
Thinking a bit more sounds good to me.

Reply:
For just accessing the cache the lock is shared. (:

Basically, it's a question of:

a) do we want to exclusive-lock the shared cache every time we access it and bump the entries immediately?
b) or do we want to only shared-lock the shared cache every time we access it and bump the entries in bulk on drop?

Both approaches do have their pros and cons.

Having it only done on drop could allow more parallelism (since the cache is, for most of the time, only shared-locked so it can be concurrently accessed by multiple readers) and it can be cheaper depending on the access patterns (imagine that the same two keys are read over and over, A, B, A, B, A, B, a million times; with the on-drop variant we only have to update their order twice, instead of a million times).

I don't have any direct evidence that one approach is better than the other, but fundamentally I think Basti had a good idea when he made it work like this.

You're definitely right that just doing an exclusive lock and just updating the order on every access would be simpler and more straightforward, although I really do like that doing it on drop allows us to minimize how long the cache is being exclusively locked.

Reply:
yes, bad eyes this morning :( (looked at the mutex from the shared access and even though I remember it was a RwLock and we register change to avoid doing write lock I thought it was the one for the shared cache).
(the mutex registering is more a `local_shared_access` than `shared_access`).
Sorry for the confusion, no pb with this logic.